### PR TITLE
Feature - Topbar redesign

### PR DESCRIPTION
**Pull request by alcar**
#### What problem is this solving?

Added:

- Store and Styles shortcuts to the admin sidebar.

Changed:

- Redesign the Site Editor's topbar and its components;
- Move the Store and Styles editors to separate admin pages;
- Move the viewport controls to the topbar.

Fixed:

- Unnecessary Y overflow on smaller viewports.

#### How should this be manually tested?

[Workspace](https://featuretopbarredesign--storecomponents.myvtex.com/admin/cms/site-editor)

#### Screenshots or example usage

Before:

image

<hr />

After:

image

#### Type of changes

| ✔️  | Type of Change                                                                            |
| --- | ----------------------------------------------------------------------------------------- |
| \_  | Bug fix                                |
| \_  | New feature                        |
| \_  | Breaking change  |
| \_  | Technical improvements  |
| ✔️  | Redesign

#### Notes

N/A.

### CODE REVIEW

File: react/components/admin/AdminStyles.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import { PageHeader } from 'vtex.styleguide'
+
+interface Props {
+  title: string
+}
+
+const AdminStyles: React.FC<Props> = ({ children, title }) => (
+  <div className="h-100 min-vh-100 overflow-y-auto bg-light-silver">
+    <div className="center mw8">
+      <PageHeader title={title} />
```
Comment (thiagomurakami) [inline react/components/admin/AdminStyles.tsx]: ⛏ `AdminStyles` is a misleading name for a component that on top of setting the container style also sets the header

File: react/components/EditorProvider.tsx [hunk 2]
```diff
@@ -19,7 +19,9 @@ import MessagesContext, { MessagesContextProps } from './MessagesContext'
 
 type Props = RenderContextProps &
   ReactIntl.InjectedIntlProps &
-  MessagesContextProps & { client: ApolloClient<unknown> }
+  MessagesContextProps & { client: ApolloClient<unknown> } & {
```
Comment (thiagomurakami) [inline react/components/EditorProvider.tsx]: ⛏ `{ client: ApolloClient<unknown>, isSiteEditor: boolean }`

File: react/components/EditorContainer/Topbar/index.tsx [hunk 3]
```diff
@@ -1,108 +1,38 @@
...
-  availableCultures,
-  iframeRuntime,
-  onChangeUrlPath,
-  urlPath,
-  visible,
-}) => {
-  const [urlInputDisabled, setUrlInputDisabled] = useState(false)
-  const [url, setUrl] = useState(urlPath)
-  const [locale, setLocale] = useState(iframeRuntime.culture.locale)
+const Topbar: React.FunctionComponent<Props> = ({ iframeRuntime }) => (
```
[the remark this answers is not in the document]
Comment (alcar) [reply react/components/EditorContainer/Topbar/index.tsx]: Thanks! 🎉

Hmm, although it's only used there, I believe that those margins are actually part of the `Topbar`, and not of the inner components. Therefore, moving those divs would mean wrongly "blackboxing" them. Do you agree?

File: react/components/admin/AdminStyles.tsx (hunk 1 above)
Comment (alcar) [reply react/components/admin/AdminStyles.tsx]: I see! I'm also not very fond of it, but had a hard time coming up with a better name...

Do you have any suggestions?

File: react/components/EditorProvider.tsx (hunk 2 above)
Comment (alcar) [reply react/components/EditorProvider.tsx]: Oops, totally missed that! Nice catch!

File: react/components/EditorContainer/Topbar/DeviceSwitcher/index.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+          <>
+            <DeviceItem
+              isActive={deviceType === editor.viewport}
+              key={deviceType}
+              onClick={handleClick}
+              position={index === 0 ? 'first' : isLast ? 'last' : 'middle'}
+              type={deviceType}
+            />
+
+            {!isLast && <div className="br c-muted-5" />}
```
[the remark this answers is not in the document]
Comment (alcar) [reply react/components/EditorContainer/Topbar/DeviceSwitcher/index.tsx]: `br` adds a 1-pixel border, while `mr1` uses a 2-pixel margin, so it's not really the same. ☹️

[the remark this answers is not in the document]
Comment (thiagomurakami) [reply react/components/EditorContainer/Topbar/DeviceSwitcher/index.tsx]: Hm, maybe extend our CSS and add a `mr05` class that has a 1px border would a way.

I just think adding a `div` to solve an issue that can be easily solved by CSS isn't the best solution.

File: react/components/EditorContainer/Topbar/index.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (thiagomurakami) [reply react/components/EditorContainer/Topbar/index.tsx]: Good point. At the same time, if those components are just a way to break down the `TopBar` into more "manageable" parts, the `div`s should actually be in the components IMO. I think it's mostly ⛏ now and you can decide if you want to change it or not!

File: react/components/admin/AdminStyles.tsx (hunk 1 above)
Comment (alcar) [reply react/components/admin/AdminStyles.tsx]: I've renamed it to `AdminStructure`.

**Pull request merged**